client/my-sites/upgrades/index.js

```diff
@@ -289,5 +290,14 @@ module.exports = function() {
 			controller.siteSelection,
 			upgradesController.checkout
 		);
+
+		page(
+			'/start-trial/:domain/:plan_name?',
```

Comment:
I also wonder if we need to get this URL whitelisted to point to Calypso, in case someone wants to link to it directly from an external source.